### ISSUE TITLE
OpenStack Image Properties are missing in >= 1.3.5

### ISSUE THREAD

**Issue by danzilio** [fog/fog#3767]
It seems that the V2 images are not returning the image properties. Here's how it looks in `1.34.0`:

```ruby
2.1.3 :015 > Fog::VERSION
 => "1.34.0"
2.1.3 :016 > image_service.images.find { |i| i.name == 'centos66-base-1.1.4' }
 =>   <Fog::Image::OpenStack::Image
    id="b51b86ae-4fab-48fc-a20b-d0d7b1a83166",
    name="centos66-base-1.1.4",
    size=4294967296,
    disk_format="ami",
    container_format="ami",
    id="b51b86ae-4fab-48fc-a20b-d0d7b1a83166",
    checksum="f2f2a39ffc564058bb52047ef65a605a",
    min_disk=0,
    created_at="2015-08-18T14:28:43",
    deleted_at=nil,
    updated_at="2015-11-04T22:32:01",
    deleted=false,
    protected=false,
    is_public=true,
    status="active",
    min_ram=0,
    owner="858769f4ed054ae387c5289b982c8952",
    properties={"kernel_id"=>"941b9291-70bb-4bae-b54a-4b51f4a78969", "ramdisk_id"=>"69c3f463-bba7-4b03-bbc7-b5e8a32b3559", "type"=>"centos66-base", "version"=>"1.1.4"},
    location=nil,
    copy_from=nil
  >
```

Here's what we're seeing in `>= 1.35.0`:

```ruby
2.1.3 :016 > Fog::VERSION
 => "1.36.0"
2.1.3 :017 > image_service.images.find { |i| i.name == 'centos66-base-1.1.4' }
 =>   <Fog::Image::OpenStack::V2::Image
    id="b51b86ae-4fab-48fc-a20b-d0d7b1a83166",
    name="centos66-base-1.1.4",
    visibility="public",
    tags=[],
    self="/v2/images/b51b86ae-4fab-48fc-a20b-d0d7b1a83166",
    size=4294967296,
    disk_format="ami",
    container_format="ami",
    id="b51b86ae-4fab-48fc-a20b-d0d7b1a83166",
    checksum="f2f2a39ffc564058bb52047ef65a605a",
    self="/v2/images/b51b86ae-4fab-48fc-a20b-d0d7b1a83166",
    file="/v2/images/b51b86ae-4fab-48fc-a20b-d0d7b1a83166/file",
    min_disk=0,
    created_at="2015-08-18T14:28:43Z",
    updated_at="2015-11-04T22:32:01Z",
    protected=false,
    status="active",
    min_ram=0,
    owner="858769f4ed054ae387c5289b982c8952",
    properties=nil,
    metadata=nil,
    location=nil,
    network_allocated=nil,
    base_image_ref=nil,
    image_type=nil,
    instance_uuid=nil,
    user_id=nil
  >
```

**Comment by geemus** [comment]
@dhague @Ladas @TerryHowe - hey, one of you maybe have some insight here? Thanks!

**Comment by Ladas** [comment]
@geemus will look at it, by coincidence I am in need of properties too, so I would probably discover it myself today :-)

**Comment by Ladas** [comment]
it works for me on gem "fog", "~>2.0.0.pre.0"

Trying 1.35

**Comment by Ladas** [comment]
@geemus hm can't update to fog 1.36 because of this
https://github.com/fog/fog/blob/bc585b8d914793aa6d4cd57ec6af7dd7d76bd203/fog.gemspec#L59

ManageIQ is using gem "fog-google", ">=0.1.1", is that fog-google downgrade in fog temporary?

@danzilio can you try if 1.36 works for you?

**Comment by danzilio** [comment]
@Ladas we're seeing this in both 1.35 and 1.36 :(

**Comment by Ladas** [comment]
@danzilio ah, seems to be actually fault of glance V2, it doesn't return the properties as a hash but as an individual attributes in the request.

so what is image.properties['architecture'] in V1 is image.attributes['architecture'] in V2, the attributes method shows all API attributes. Only some of them are usually exposed as attributes of model. Meaning for v1 thse are equivalent.

image.properties['architecture'] == image.attributes['properties']['architecture']

So for v2, just use image.attributes['architecture'], unless Glance team realizes it's bug, since it's not documented like this. :-). Only these should be supported in glance http://docs.openstack.org/cli-reference/content/chapter_cli-glance-property.html

My code will contain something like this to cover both V1 and V2

architecture = image.properties.try(:[], 'architecture') || image.attributes['architecture']

**Comment by danzilio** [comment]
Ahh! That makes more sense now! Thanks @Ladas!

**Comment by geemus** [comment]
The fog-google downgrade/restriction is because fog-google after that is no longer 1.9 compatible and we are trying to (for the time being) keep the rest of fog 1.9.

**Comment by Ladas** [comment]
@geemus hm, will that apply also for 2.0? Trying to figure out how to unblock Fog for ManageIQ, we are having Google provider so they need the newest google gem.

I guess for 2.0 just documenting the fact and letting people to limit their fog-google in their GemFile if they want 1.9 compat?

**Comment by geemus** [comment]
@Ladas I'm not sure. Originally we were targeting 2.0 to just drop 1.8, but 1.9 got added as a possibility due to net-ssh (and now mime-types). For the time being I'm still hoping to support 1.9 for most things in the near term. It may be that users will be better off using the already divided gems in order to avoid and manage this themselves (and apply further pressure to subdivide the remaining ones out of fog/fog).

**Comment by Ladas** [comment]
@geemus right, so starting fog-openstack that would support just ruby 2 would be the way? Or it should be in  fog-openstack-core? 

Though if it's needed for other gems, I don't have problem dropping 1.9, given 1.9 and 2.0 have like 99% compatibility, people should not have problem moving, right? 

Should I start another issue for this? :-)

**Comment by geemus** [comment]
@Ladas splitting it out seems like a win regardless. We definitely would like to drop 1.8 support ASAP, but I think 1.9 may linger a bit longer still (due to some OSes having old versions and things). Another issue might be a good idea.

**Comment by Ladas** [comment]
@geemus https://github.com/fog/fog/issues/3778 it's kinda big issue for us now, cause we cannot move to newer version of Fog. Some quick and painless solution would be the best. :-)

**Comment by geemus** [comment]
got it. I mentioned the fog-google maintainers on the other thread, so maybe they'll have more insight/thoughts.

**Comment by Ladas** [comment]
@geemus awesome, thank you. I think we can close this one :-)

**Comment by geemus** [comment]
@Ladas great, thanks.
